client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java

```diff
@@ -0,0 +1,201 @@
+    Assert.assertEquals("Wrong life cycle executor maximum pool size", DEFAULT_MAX_THREAD_IDLE_MILLISECONDS, maxThreadIdleMilliseconds);
+
+    executorService = executorContext.getApiExecutor();
+    Assert.assertTrue(executorService instanceof ThreadPoolExecutor);
+    threadPoolExecutor = (ThreadPoolExecutor) executorService;
+    corePoolSize = threadPoolExecutor.getCorePoolSize();
+    maxPoolSize = threadPoolExecutor.getMaximumPoolSize();
+    maxThreadIdleMilliseconds = threadPoolExecutor.getKeepAliveTime(TimeUnit.MILLISECONDS);
+    Assert.assertEquals("Wrong api executor core pool size", DEFAULT_CORE_POOL_SIZE, corePoolSize);
+    Assert.assertEquals("Wrong api executor maximum pool size", CUSTOM_API_THREAD_COUNT, maxPoolSize);
+    Assert.assertEquals("Wrong api executor maximum pool size", DEFAULT_MAX_THREAD_IDLE_MILLISECONDS, maxThreadIdleMilliseconds);
+
+    executorService = executorContext.getCallbackExecutor();
+    Assert.assertTrue(executorService instanceof ThreadPoolExecutor);
+    threadPoolExecutor = (ThreadPoolExecutor) executorService;
+    corePoolSize = threadPoolExecutor.getCorePoolSize();
+    maxPoolSize = threadPoolExecutor.getMaximumPoolSize();
+    maxThreadIdleMilliseconds = threadPoolExecutor.getKeepAliveTime(TimeUnit.MILLISECONDS);
+    Assert.assertEquals("Wrong callback executor core pool size", DEFAULT_CORE_POOL_SIZE, corePoolSize);
+    Assert.assertEquals("Wrong callback executor maximum pool size", CUSTOM_CALLBACK_THREAD_COUNT, maxPoolSize);
+    Assert.assertEquals("Wrong callback executor maximum pool size", DEFAULT_MAX_THREAD_IDLE_MILLISECONDS, maxThreadIdleMilliseconds);
+
+    executorService = executorContext.getScheduledExecutor();
+    Assert.assertTrue(executorService instanceof ScheduledThreadPoolExecutor);
+    threadPoolExecutor = (ScheduledThreadPoolExecutor) executorService;
+    corePoolSize = threadPoolExecutor.getCorePoolSize();
+    Assert.assertEquals("Wrong scheduled executor core pool size", CUSTOM_SCHEDULED_THREAD_COUNT, corePoolSize);
+  }
+
+  @Test
+  public void testConstructorWithBuilder() throws Exception {
+    executorContext = new FlexibleExecutorContext.FlexibleExecutorContextBuilder()
+        .setLifeCycleThreadCount(CUSTOM_LIFE_CYCLE_THREAD_COUNT)
+        .setMaxLifeCycleThreadsIdleMilliseconds(CUSTOM_MAX_LIFE_CYCLE_THREADS_IDLE_MILLISECONDS)
+        .setApiThreadCount(CUSTOM_API_THREAD_COUNT)
+        .setMaxApiThreadsIdleMilliseconds(CUSTOM_MAX_API_THREADS_IDLE_MILLISECONDS)
+        .setCallbackThreadCount(CUSTOM_CALLBACK_THREAD_COUNT)
+        .setMaxCallbackThreadsIdleMilliseconds(CUSTOM_MAX_CALLBACK_THREADS_IDLE_MILLISECONDS)
+        .setScheduledThreadCount(CUSTOM_SCHEDULED_THREAD_COUNT)
+        .build();
+    executorContext.init();
+
+    ExecutorService executorService;
+    ThreadPoolExecutor threadPoolExecutor;
+    long corePoolSize, maxPoolSize, maxThreadIdleMilliseconds;
```

Comment:
Use **int** type as in constants.
Declare every variable in separate line. No variable grouping is recommended.